test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Metadata/CaseInsensitiveSchemaElementsCacheTests.cs

```diff
@@ -0,0 +1,166 @@
+            model.AddElement(func1);
+            model.AddElement(func2);
+
+            var cache = new CaseInsensitiveSchemaElementsCache(model);
+
+            var matches = cache.FindOperations("ns.models.computeStuff");
+            Assert.Equal(2, matches.Count);
+            Assert.Contains(matches, match => match == func1);
+            Assert.Contains(matches, match => match == func2);
+        }
+
+        [Fact]
+        public void FindOperations_ReturnsNullIfThereAreNoMatches()
+        {
+            var model = new EdmModel();
+            var operation = new EdmAction("NS.Models", "DoStuff", EdmCoreModel.Instance.GetBoolean(false));
+            model.AddElement(operation);
+            model.AddElement(new EdmFunction("Ns.Models", "ComputeStuff", EdmCoreModel.Instance.GetBoolean(false)));
+
+            var cache = new CaseInsensitiveSchemaElementsCache(model);
+
+            var matches = cache.FindOperations("ns.models.dost");
+            Assert.Null(matches);
+        }
+
+        [Theory]
+        [InlineData("NS.Models.SomeTerm")]
+        [InlineData("ns.models.someterm")]
+        [InlineData("ns.MODELS.someTerm")]
+        public void FindTerm_FindsTermsBasedOnCaseInsensitiveMatch(string qualifiedName)
+        {
+            var model = new EdmModel();
+            var term = model.AddTerm("NS.Models", "SomeTerm", EdmPrimitiveTypeKind.String);
+            model.AddTerm("NS.Models", "OtherTerm", EdmPrimitiveTypeKind.String);
+
+            var cache = new CaseInsensitiveSchemaElementsCache(model);
+
+            var matches = cache.FindTerms(qualifiedName);
+
+            Assert.Single(matches);
+            Assert.Equal(term, matches[0]);
+        }
+
+        [Fact]
+        public void FindTerms_FindsAllTermshatMatchName()
+        {
+            var model = new EdmModel();
+            var term1 = model.AddTerm("NS.Models", "SomeTerm", EdmPrimitiveTypeKind.String);
+            model.AddTerm("NS.Models", "OtherTerm", EdmPrimitiveTypeKind.String);
+            var term2 = model.AddTerm("NS.Models", "someTerm", EdmPrimitiveTypeKind.String);
+
+            var cache = new CaseInsensitiveSchemaElementsCache(model);
+
+            var matches = cache.FindTerms("ns.models.someTerm");
+            Assert.Equal(2, matches.Count);
+            Assert.Contains(matches, match => match == term1);
+            Assert.Contains(matches, match => match == term2);
+        }
+
+        [Fact]
+        public void FindTerms_ReturnsNullIfThereAreNoMatches()
+        {
+            var model = new EdmModel();
+            model.AddTerm("NS.Models", "SomeTerm", EdmPrimitiveTypeKind.String);
+            model.AddTerm("NS.Models", "OtherTerm", EdmPrimitiveTypeKind.String);
+
+            var cache = new CaseInsensitiveSchemaElementsCache(model);
+
+            var matches = cache.FindTerms("ns.models.some");
+            Assert.Null(matches);
+        }
+    }
+}

```

Comment:
Add a test for a scenario where the model passed to the `CaseInsensitiveSchemaElementsCache` constructor is null